interface/weno/formularyInsert.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * weno rx formulary update.
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @copyright Copyright (c) 2016-2017 Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+ require_once("../globals.php");
+ 
+ echo xlt("Trying")."<BR>";
+
+$zip = new ZipArchive;
+$zip->open('FormularyFiles.zip');
+$zip->extractTo($GLOBALS['temporary_files_dir']);
+$zip->close();
+
+ $file = $GLOBALS['temporary_files_dir'] . "./Weno Exchange LLC/BSureFormulary.txt";
+ $lines = count(file($file));
+ echo text($lines). "<br>";
+  
+ $i = 2;
+
+do {
+    $fileName = new SplFileObject($file);
+    $fileName->seek($i);
+ 
+    $in = explode("|", $fileName);
+
+    $ndc = $in[2];
+    $price = $in[5];
+    $drugName = $in[8];
+
+    sqlStatementNoLog("INSERT INTO `erx_drug_paid` SET `drug_label_name` = ?, `NDC` = ?, `price_per_unit` = ? ", array($drugName,$ndc,$price));
+    echo xlt("Inserted") . " " . text($drugName) . "<br> ";
+
+    $i++;
+} while ($i < $lines);
```

Comment:
below above while code, do the following:
``````
// Settings to drastically speed up import with InnoDB
sqlStatementNoLog("COMMIT");
sqlStatementNoLog("SET autocommit=1");

Reply:
I usually just do what you ask no questions and try to understand later. Sometimes it is obvious after I read the instructions and I add it to my skills. 
This one is over my head enough that I want to ask a question. Placing the statements after the while loop means they will only execute after the loop is finished. How is that going to speed up the insert process?

Reply:
In this case, we are wrapping the large insert with settings to increase the innodb performance. Note you also need to place the settings above the while loop in my other post above. For inserts, innodb touches the hard drive after every insert for log/robust issues. This can be very expensive when inserting hundreds of records(for example, the setup script in openemr could take up to 45 minutes because of 100,000+ translations), so we are basically turning this off temporarily before the insert, and then turning it back on after the insert.

Reply:
understood.